src/main/fc/core.c

```diff
@@ -979,10 +986,27 @@ void processRxModes(timeUs_t currentTimeUs)
         DISABLE_FLIGHT_MODE(ANGLE_MODE); // failsafe support
     }
 
-    if (IS_RC_MODE_ACTIVE(BOXHORIZON) && canUseHorizonMode) {
+#ifdef USE_ALT_HOLD_MODE
+    // only if armed
+    if (ARMING_FLAG(ARMED) 
+    // and either the alt_hold switch is activated, or are in failsafe
+    && (IS_RC_MODE_ACTIVE(BOXALTHOLD) || failsafeIsActive())
+    // but not in GPS_RESCUE_MODE, ie if failsafe is active, must be in Landing Mode
+    && (!FLIGHT_MODE(GPS_RESCUE_MODE))
+    // and we have Acc for self-levelling
+    && (sensors(SENSOR_ACC))
+    // and we have already taken off (to prevent activation on the ground), then enable althold
+    && isAirmodeActivated()) {
```

Comment:
```suggestion
    if (ARMING_FLAG(ARMED) 
        // and either the alt_hold switch is activated, or are in failsafe
        && (IS_RC_MODE_ACTIVE(BOXALTHOLD) || failsafeIsActive())
        // but not in GPS_RESCUE_MODE, ie if failsafe is active, must be in Landing Mode
        && !FLIGHT_MODE(GPS_RESCUE_MODE)
        // and we have Acc for self-levelling
        && sensors(SENSOR_ACC)
        // and we have already taken off (to prevent activation on the ground), then enable althold
        && isAirmodeActivated()) {
```

Indentation is to `ARMING_FLAG(ARMED)` in 'if ('. 

It coincides with block indent - 4 spaces - which is pretty unfortunate. Maybe adding 2 spaces into `if (  ARMIN`. Or use 2 space indent and move ARMING_FLAG(ARMED) to next line.

Reply:
I changed to simple indenting, perhaps it might be OK like that.